### PR TITLE
Update and add tests for distributions

### LINKED ISSUE #72: Our Gaussian class could subclass torch.distributions.Normal
We could get rid of some bulk like the normalisations etc.

### PR DESCRIPTION
**Pull request by wilsonmr**
closes #72 and also follows the general policy of increasing coverage of docs and tests

### CODE REVIEW

File: anvil/distributions.py [hunk 1]
```diff
@@ -24,40 +22,45 @@ class Gaussian:
         Mean for the distribution.
     """
 
-    def __init__(self, size_out, *, sigma=1, mean=0):
+    def __init__(self, size_out, *, mean=0, sigma=1):
+        super().__init__(mean, sigma)
```
Comment (jmarshrossney) [inline anvil/distributions.py]: Do you think we should just follow PyTorch by labelling these parameters `loc` and `scale`? This would be more consistent with the way in which we wrap around `torch.optim` (i.e. not modifying the labelling of hyperparameters)

File: anvil/distributions.py [hunk 2]
```diff
@@ -82,37 +85,41 @@ class PhiFourScalar:
...
+    >>> from anvil.geometry import Geometry2D
+    >>> from anvil.distributions import PhiFourScalar
+    >>> import torch
     >>> geom = Geometry2D(2)
-    >>> action = PhiFourAction(geom, "standard", {"m_sq": 4, "g": 0})
-    >>> state = torch.rand((1, 2*2))
-    >>> action(state)
+    >>> action = PhiFourScalar.from_standard(geom, **{"m_sq": 4, "g": 0})
+    >>> state = torch.rand((1, 2*2)) # 2-D so lattice cardinality is 4
+    >>> action.log_density(state)
```
Comment (jmarshrossney) [inline anvil/distributions.py]: I know it's pretty anal but personally I'd replace `action` with `target` or something similar, just because `action.log_density` is a bit confusing.

**Review by jmarshrossney — APPROVED** [review]

**Comment by jmarshrossney** [comment]
Happy to make the very minor changes suggested above, let me know what you think.

**Comment by wilsonmr** [comment]
Yep go for it

**Comment by jmarshrossney** [comment]
Couldn't bring myself to change `sigma` to `scale` in the config parser - it'd probably do more harm than good!

**Comment by wilsonmr** [comment]
yeah the "idea" with these things is the high level interface is very well thought out and never changes (so the user can keep using their fantastic runcards) and the lower level interface can change with the wind.

In reality you always have to make these kinds of choices.

**Comment by wilsonmr** [comment]
haha, oops - forgot about the tests which check the low level interface 😄

**Comment by jmarshrossney** [comment]
crap..
